extensions/ros2/launch

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+function source_with_prefix() {
+    export COLCON_CURRENT_PREFIX="$1"
+    if [ -f "$COLCON_CURRENT_PREFIX/local_setup.bash" ]; then
+      # shellcheck disable=SC1090
+      source "$COLCON_CURRENT_PREFIX/local_setup.bash"
+    else
+      echo "warning: no setup found for $COLCON_CURRENT_PREFIX"
```

Comment:
This would almost certainly be an error, unless this is needed at pull time before it's in place?

Reply:
This is strictly for run-time. I didn't error on it because the v1 script ignores it if it doesn't exist, and I thought there may be case where it didn't exist.  So I just warned.  I'll instead error :+1:

Reply:
Yeah make it error. The other one had to deal with it not being there because `env()` was evaluated for pulling, before it was actually present. This is much nicer now.

Reply:
done